net/src/packet.rs

```diff
@@ -15,22 +16,25 @@ use crate::parse::{
     Writer,
 };
 use crate::tcp::Tcp;
-use crate::udp::Udp;
-use crate::vlan::Vlan;
+use crate::udp::{Encap, Udp};
+use crate::vlan::{Pcp, Vid, Vlan};
+use crate::vxlan::Vxlan;
 use arrayvec::ArrayVec;
 use std::num::NonZero;
 use tracing::debug;
 
 const MAX_VLANS: usize = 4;
 const MAX_NET_EXTENSIONS: usize = 2;
 
-#[derive(Debug, Clone, PartialEq, Eq)]
+// TODO: remove `pub` from all fields
+#[derive(Debug)]
 pub struct Packet {
```

Comment:
I'd reorder the fields here to reflect the order in which they are populated: eth/vlan/net ....
It's mostly cosmetic, yes.
Also, for the encap, would it make sense to nest it inside transport?

Reply:
I debated a lot about encap being inside transport or not.

In the end I suggest that we keep it outside of transport because putting it in the transport variant increases the size of the `Udp` (et al) object and that is a very useful struct on its own.

Reply:
How about the ordering?

Reply:
That is also a fair point.  There may be some very tiny advantage in terms of locality in organizing the structure to include the most likely to be used fields close together, but that is pure speculation and does not really justify the deviation from the OSI layer mental model (at least not in the absence of strong performance numbers)